execution.bs

```diff
@@ -6493,142 +6497,353 @@ template&lt;class Domain, class Tag, sender Sndr, class... Args>
-                      all of the operation states in `sh_state`'s list of operation
-                      states are notified as if `set_stopped(rcvr)` had
-                      been called.
-
-                  * Otherwise, `start(op_state2)` is called.
-
-        7. When `rcvr` completes it will notify `op_state` that the result are
-            ready. Let <code><i>Tag</i></code> be whichever
-            completion function was called on receiver `rcvr`. `op_state`'s
-            stop callback optional is reset. Then
-            <code><i>Tag</i>(std::move(out_rcvr2), args2...)</code> is called,
-            where `args2...` is a pack of const lvalues referencing the subobjects of
-            `sh_state` that have been saved by the original call to
-            <code><i>Tag</i>(rcvr, args...)</code>.
-
-        8. Ownership of `sh_state` is shared by `sndr2` and by every `op_state`
-            that results from connecting `sndr2` to a receiver.
-
-    2. Given subexpressions `sndr2` where `sndr2` is a sender returned from `split`
-        or a copy of such, `get_env(sndr2)` shall return an lvalue reference to the
-        object in `sh_state` that was initialized with the result of `get_env(sndr)`.
-
-5. Let `sndr` be a sender expression, `rcvr` be an instance of the receiver type
-    described above, `sndr2` be a sender returned from `split(sndr)` or a copy of
-    such, `rcvr2` is the receiver to which `sndr2` is connected, and `args` is the
-    pack of subexpressions passed to `rcvr`'s completion function
-    <code><i>CSO</i></code> when `sndr` completes. `sndr2` shall invoke
-    <code><i>CSO</i>(rcvr2, args2...)</code> where `args2` is a pack of const
-    lvalue references to objects decay-copied from `args`, or by calling
-    <code>set_error(rcvr2, err2)</code> for some subexpression `err2`. The objects
-    passed to `rcvr2`'s completion operation shall be valid until after the
-    completion of the invocation of `rcvr2`'s completion operation.
+        1. *Effects:* Equivalent to:
+
+            <pre highlight="c++">
+            on_stop.reset();
+            visit(
+              [this]&lt;class Tuple>(Tuple&& tupl) noexcept -> void {
+                apply(
+                  [this](auto tag, auto&... args) noexcept -> void {
+                    tag(std::move(*rcvr), std::forward_like&lt;Tuple>(args)...);
+                  },
+                  tupl);
+              },
+              <i>QUAL</i>(sh_state->result));
+            )
+            </pre>
+
+            where <i>`QUAL`</i> is `std::move` if `tag_of_t<Sndr>` is
+            <i>`ensure-started-impl-tag`</i>, and `as_const` otherwise.
+
+    5. <pre highlight="c++">
+        void <i>detach</i>() noexcept override;</pre>
+
+        1. *Effects:* Equivalent to <code>sh_state-><i>detach</i>()</code>
+            if `tag_of_t<Sndr>` is <i>`ensure-started-impl-tag`</i>; otherwise,
+            nothing.
```

Comment:
```suggestion
        1. *Effects:* Equivalent to <code>sh_state-><i>detach</i>()</code>
            if same_as<`tag_of_t<Sndr>`, <i>`ensure-started-impl-tag`</i> > is `true`; otherwise,
            nothing.
```